In [1]:
#usaremos a biblioteca pandas
import pandas as pd

Base de dados com os 25 esportistas mais bem pagos segundo a forbes, para 2023. Os dados dos ganhos são em Reais e na casa dos milhões por ano.



In [4]:
df = pd.read_csv("ESPORTISTAS_MOD7.csv", delimiter=';')

df.head(10)

Atleta            Esporte  Ganhos_Totais
0       Lionel Messi            Futebol            639
1       Lebron James           Basquete            595
2  Cristiano Ronaldo            Futebol            565
3          Neymar Jr            Futebol            467
4      Stephen Curry           Basquete            452
5        Kevin Duart           Basquete            452
6      Roger Federer              Tenis            442
7     Canelo Alvarez               Boxe            442
8          Tom Brady  Futebol Americano            408
9   Giannis Antetoku           Basquete            397

Conhecendo a base.

In [ ]:
# Usando o group by para agrupar os dados do df pela coluna 'Esporte'
df.groupby('Esporte')['Atleta'].count().reset_index().sort_values(by='Atleta', ascending=False)

Esporte  Atleta
1           Basquete       8
5  Futebol Americano       7
4            Futebol       3
3               Boxe       2
7              Tenis       2
0      Automobilismo       1
2           Beisebol       1
6              Golfe       1

25 atletas 8 deles são de Basquete

In [7]:
# A porcentagem de cada esporte
contagem_por_esporte = df['Esporte'].value_counts(normalize=True).reset_index()
contagem_por_esporte

Esporte  proportion
0           Basquete        0.32
1  Futebol Americano        0.28
2            Futebol        0.12
3              Tenis        0.08
4               Boxe        0.08
5              Golfe        0.04
6      Automobilismo        0.04
7           Beisebol        0.04

# Aplicando os conceitos de estatística:

## 1 - MÉDIA
De forma manual

In [8]:
# Criando uma variavel chamada soma_ganhos_totais e em seguida aplicamos o SUM para que seja feita a soma da coluna.
soma_ganhos_totais = df['Ganhos_Totais'].sum()
# Em seguida calculando a quantidade de atletas 
quantidade_de_registros = len(df['Ganhos_Totais'])
# Dividimos a soma da coluna de ganhos pela quantidade de atletas
media_ganhos_totais = soma_ganhos_totais / quantidade_de_registros

print("Média de ganhos totais:", media_ganhos_totais)

Média de ganhos totais: 377.0


Com a função mean:

In [9]:
df['Ganhos_Totais'].mean()

377.0

Ganhos totais por esporte utilizando mean:

In [ ]:
df.groupby('Esporte')['Ganhos_Totais'].mean().reset_index().sort_values(by='Ganhos_Totais', ascending=False)

Esporte  Ganhos_Totais
4            Futebol          557.0
1           Basquete          400.5
3               Boxe          373.5
7              Tenis          366.0
6              Golfe          334.0
0      Automobilismo          319.0
5  Futebol Americano          311.0
2           Beisebol          241.0

## 2 - MEDIANA

Com a função median que ao ser aplicada funciona exatamente igual a média.  

In [11]:
df['Ganhos_Totais'].median()

354.0

A mediana é menor que média, que foi de 377.
Isso evidencia uma distribuição dos dados não tão uniforme pois alguns valores tem puxado a média para um valor acima da mediana.

In [12]:
df.groupby('Esporte')['Ganhos_Totais'].median().reset_index().sort_values(by='Ganhos_Totais', ascending=False)

Esporte  Ganhos_Totais
4            Futebol          565.0
1           Basquete          392.5
3               Boxe          373.5
7              Tenis          366.0
6              Golfe          334.0
5  Futebol Americano          329.0
0      Automobilismo          319.0
2           Beisebol          241.0

## 3 - MODA

A moda serve para ver um total de ganhos que se repete mas não é uma métrica tão eficiente para análise em específico.

In [13]:
df['Ganhos_Totais'].mode()

0    241
1    273
2    334
3    442
4    452
Name: Ganhos_Totais, dtype: int64

## 4 - DESVIO PADRÃO

Ao comparar a média com a mediana nota-se que temos uma distribuição de dados não tao uniforme, através do desvio pode-se identificar qual dos esportes tem maior dispersão entre os ganhos

In [14]:
desvio_padrao_por_esporte = df.groupby('Esporte')['Ganhos_Totais'].std().reset_index()
desvio_padrao_por_esporte

Esporte  Ganhos_Totais
0      Automobilismo            NaN
1           Basquete     103.414561
2           Beisebol            NaN
3               Boxe      96.873629
4            Futebol      86.278618
5  Futebol Americano      62.960305
6              Golfe            NaN
7              Tenis     107.480231

Os registros estão NAN são de quando tem apenas 1 registro para o esporte, pois não temos como calcular o desvio padrão se não temos mais de um registro.

In [15]:
df.describe()

Ganhos_Totais
count      25.000000
mean      377.000000
std       110.029541
min       238.000000
25%       290.000000
50%       354.000000
75%       442.000000
max       639.000000